Cython/Compiler/ExprNodes.py

```diff
@@ -2910,8 +2840,107 @@ def free_temps(self, code):
         if self.iter_func_ptr:
             code.funcstate.release_temp(self.iter_func_ptr)
             self.iter_func_ptr = None
-        if self.cpp_iterator_cname:
-            code.funcstate.release_temp(self.cpp_iterator_cname)
+        ExprNode.free_temps(self, code)
+
+
+class CppIteratorNode(IteratorNode):
```

Comment:
Does it need to inherit from `IteratorNode`? `AsyncIteratorNode` is a simple `ExprNode`, for example.